auth/auth.go

```diff
@@ -101,6 +107,33 @@ func (a ACL) VerifyPeerCertificateServer(rawCerts [][]byte, verifiedChains [][]*
 		return nil
 	}
 
+	input := map[string]interface{}{
+		"certificate": cert,
+	}
+
+	// Check against OPA
+	if a.AllowOPAQuery != nil {
+		results, err := a.AllowOPAQuery.Eval(context.TODO(), rego.EvalInput(input))
```

Comment:
Let's thread this context from the proxy module. In the proxy, each new connection should create a context and then pass it along to here. This way, we can set a deadline for new connections in the proxy module in the future to avoid hanging if a policy takes too long to execute.

Reply:
A good default timeout for querying the policy is probably around 1s, what do you think?

Reply:
Maybe we could re-use the connect timeout, but I just realized threading the timeout through might be tricky because this is in a callback. So maybe instead of threading a context through we can just put the connect timeout on the eval directly

Reply:
Ok - makes sense. I will wire a timeout on the eval then.